libs/ngxtension/merge-from/src/lib/merge-from.ts

```diff
@@ -0,0 +1,114 @@
+import { Injector, isSignal, untracked, type Signal } from '@angular/core';
+import { toObservable, toSignal } from '@angular/core/rxjs-interop';
+import { assertInjector } from 'ngxtension/assert-injector';
+import {
+	distinctUntilChanged,
+	from,
+	isObservable,
+	merge,
+	startWith,
+	type ObservableInput,
+	type OperatorFunction,
+} from 'rxjs';
+
+export type ObservableSignalInput<T> = ObservableInput<T> | Signal<T>;
+
+/**
+ * So that we can have `fn([Observable<A>, Signal<B>]): Observable<[A, B]>`
+ */
+type ObservableSignalInputTuple<T> = {
+	[K in keyof T]: ObservableSignalInput<T[K]>;
+};
+
+export type MergeFromOptions<T> = {
+	readonly injector?: Injector;
+	readonly initialValue?: T | null;
+};
+
+export function mergeFrom<
+	Inputs extends readonly unknown[],
+	Output = Inputs[number],
+>(
+	inputs: readonly [...ObservableSignalInputTuple<Inputs>],
+	operator?: OperatorFunction<Inputs[number], Output>,
+	options?: MergeFromOptions<Output>,
+): Signal<Output>;
+export function mergeFrom<
+	Inputs extends readonly unknown[],
+	Output = Inputs[number],
+>(
+	inputs: readonly [...ObservableSignalInputTuple<Inputs>],
+	options?: MergeFromOptions<Output>,
+): Signal<Output>;
+
+export function mergeFrom<
+	Inputs extends readonly unknown[],
+	Output = Inputs[number],
+>(...args: unknown[]) {
+	const [sources, operator, options = {}] = parseArgs<Inputs, Output>(args);
+	const normalizedSources = sources.map((source) => {
+		if (isSignal(source)) {
+			return toObservable(source, { injector: options.injector }).pipe(
+				startWith(untracked(source)),
+			);
+		}
+
+		if (!isObservable(source)) {
+			source = from(source);
+		}
+
+		return source.pipe(distinctUntilChanged());
+	});
+
+	const merged = operator
+		? merge(...normalizedSources).pipe(operator)
+		: merge(...normalizedSources);
```

Comment:
You can simplify this with the `identity` operator from rxjs
```suggestion
	const merged = merge(...normalizedSources).pipe(operator ?? identity);
```

Reply:
Or you could even remove `?? identity` if `identity` is the default value when you destrucure the result from `parseArgs`

Reply:
Thanks